src/factory/DerivedMeasureTitleSuffixFactory.ts

```diff
@@ -22,26 +22,25 @@ export default class DerivedMeasureTitleSuffixFactory {
     }
 
     /**
-     * Returns formatted localized suffix string for derived measure based on the measure definition type.
-     * In case when simple measure definition is provided the empty string is returned.
+     * Returns formatted localized suffix string for derived measure based on the over time comparison type.
+     * In case when unsupported over time comparison type is provided the empty string is returned.
      *
-     * @param {VisualizationObject.IMeasureDefinitionType} measureDefinition - The measure definition for which the
+     * @param {OverTimeComparisonType} overTimeComparisonType - The over time comparison type for which the
      *      suffix must be obtained.
      * @returns {string}
      */
-    public getSuffix(measureDefinition: IMeasureDefinitionType): string {
-        const localizationKey = this.getSuffixLocalizationKey(measureDefinition);
+    public getSuffix(overTimeComparisonType: OverTimeComparisonType): string {
+        const localizationKey = this.getSuffixLocalizationKey(overTimeComparisonType);
         return localizationKey === null ? '' : ` - ${this.translateKey(localizationKey)}`;
     }
 
-    private getSuffixLocalizationKey(measureDefinition: IMeasureDefinitionType): string {
-        if (VisualizationObject.isPopMeasureDefinition(measureDefinition)) {
+    private getSuffixLocalizationKey(overTimeComparisonType: OverTimeComparisonType): string {
+        if (overTimeComparisonType === OverTimeComparisonTypes.SAME_PERIOD_PREVIOUS_YEAR) {
```

Comment:
this looks to me like a good place for using `switch-case` what do you think ? :)

Reply:
Fixed